waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/metrics/MonitoringConfiguration.java

```diff
@@ -15,102 +15,86 @@
-            ((ScheduledReporter) reporter).report();
-          }
-          reporter.close();
-        } catch (Exception e) {
-          LOG.warn("Problem closing reporter", e);
-        }
+      @Override
+      public String host() {
+        return graphiteConfiguration.getHost();
       }
-    }
-  }
 
-  @PostConstruct()
-  public void init() {
-    registerBaseMetrics();
-    registerReporter(jmxReporterBuilder().build()).start();
-    if (graphiteConfiguration.isEnabled()) {
-      GraphiteReporter graphiteReporter = graphiteReporterBuilder().build(newGraphite());
-      registerReporter(graphiteReporter);
-      graphiteReporter.start(graphiteConfiguration.getPollInterval(), graphiteConfiguration.getPollIntervalTimeUnit());
-    }
-  }
+      @Override
+      public int port() {
+        return graphiteConfiguration.getPort();
+      }
 
-  private void registerBaseMetrics() {
-    metricRegistry.register("gc", new GarbageCollectorMetricSet());
-    metricRegistry.register("memory", new MemoryUsageGaugeSet());
-    metricRegistry.register("threads", new ThreadStatesGaugeSet());
-  }
+      @Override
+      public boolean enabled() {
+        return graphiteConfiguration.isEnabled();
+      }
 
-  private JmxReporter.Builder jmxReporterBuilder() {
-    return JmxReporter.forRegistry(metricRegistry);
-  }
+      @Override
+      public String[] tagsAsPrefix() {
+        return new String[] { graphiteConfiguration.getPrefix() };
+      }
 
-  private GraphiteReporter.Builder graphiteReporterBuilder() {
-    return GraphiteReporter.forRegistry(metricRegistry).convertRatesTo(TimeUnit.SECONDS)
-        .convertDurationsTo(TimeUnit.MILLISECONDS).filter(MetricFilter.ALL)
-        .prefixedWith(graphiteConfiguration.getPrefix());
-  }
+      @Override
+      public TimeUnit durationUnits() {
+        return graphiteConfiguration.getPollIntervalTimeUnit();
+      }
 
-  private Graphite newGraphite() {
-    return new Graphite(new InetSocketAddress(graphiteConfiguration.getHost(), graphiteConfiguration.getPort()));
-  }
+      @Override
+      public GraphiteProtocol protocol() {
+        return GraphiteProtocol.PLAINTEXT;
+      }
 
-  Set<Closeable> getReporters() {
-    return reporters;
-  }
+      @Override
+      public String get(String arg0) {
+        return null;
+      }
+    };
+
+    MeterRegistry graphiteMeterRegistry = null;
+    if (graphiteConfiguration.isEnabled()) {
+      HierarchicalNameMapper wdHierarchicalNameMapper = (id, convention) -> graphiteConfiguration.getPrefix()
+          + "."
+          + HierarchicalNameMapper.DEFAULT.toHierarchicalName(id, convention);
 
-  void setGraphiteConfiguration(GraphiteConfiguration graphiteConfiguration) {
-    this.graphiteConfiguration = graphiteConfiguration;
+      graphiteMeterRegistry = new GraphiteMeterRegistry(graphiteConfig, Clock.SYSTEM, wdHierarchicalNameMapper);
+
+      graphiteMeterRegistry.config().namingConvention(NamingConvention.dot);
```

Comment:
I thought dot was the default convention but if you're doing this then I assume that's not the case?

Reply:
Yeah. Annoyingly, CamelCase is now default in GraphiteNamingConvention.

https://github.com/micrometer-metrics/micrometer/blob/master/implementations/micrometer-registry-graphite/src/main/java/io/micrometer/graphite/GraphiteNamingConvention.java#L35